backend/yubikey.go

```diff
@@ -0,0 +1,206 @@
+package backend
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"github.com/foxboron/sbctl/config"
+	"github.com/foxboron/sbctl/logging"
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/tomis007/piv-go/v2/piv"
+)
+
+type Yubikey struct {
+	keytype BackendType
+	cert    *x509.Certificate
+	privkey *piv.KeyRSA
```

Comment:
```suggestion
	privkey crypto.Signer
```

Which is why I suspect the upstream author disagrees. This also generalizes the support for RSA and NIST keys.

Reply:
I write while recalling that Secure Boot only support RSA keys :)

Reply:
Ah yes I see the issue, I was unable to figure out type casting the `priv` to crypto.Signer. Updated to fix that and to not use my fork